File: backend/ModuMessenger/src/main/java/com/example/modumessenger/chat/dto/ChatDto.java
package com.example.modumessenger.chat.dto;

import com.example.modumessenger.chat.entity.Chat;
import com.example.modumessenger.chat.entity.ChatRoom;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatDto implements Serializable {
    private int chatType;
    private String roomId;
    private String sender;
    private String message;
    private String chatTime;
    private ChatRoomDto chatRoomDto;

    public ChatDto(String msg, String roomId, String sender, String chatTime, int type, ChatRoomDto chatRoomDto) {
        this.message = msg;
        this.roomId = roomId;
        this.sender = sender;
        this.chatTime = chatTime;
        this.chatType = type;
        this.chatRoomDto = chatRoomDto;
    }
}

File: backend/chat-store-service/src/main/kotlin/com/example/chatstoreservice/chat/dto/ChatDto.kt
package com.example.chatstoreservice.chat.dto

import java.io.Serializable

/** chat-service 내부 API 응답. 모든 필드에 기본값이 있어 Jackson 이 기본 생성자로 만든다. */
data class ChatDto(
    var id: Long? = null,
    var chatType: Int = 0,
    var roomId: String? = null,
    var sender: String? = null,
    var message: String? = null,
    var chatTime: String? = null,
    var chatRoomDto: ChatRoomDto? = null,
) : Serializable
